client/types.go

```diff
@@ -0,0 +1,39 @@
+package client
+
+import (
+	"github.com/iov-one/weave"
+	cmn "github.com/tendermint/tendermint/libs/common"
+)
+
+// TransactionID is the hash used to identify the transaction
+type TransactionID = cmn.HexBytes
+
+// TxQuery is some query to find transactions
+type TxQuery string
+
+// MempoolResult is returned from the mempool (CheckTx)
+// Result is only set on success codes, Err is set if it was a failure code
+type MempoolResult struct {
+	ID  TransactionID
+	Err error
+}
+
+// AsCommitError will turn an errored MempoolResult into a CommitResult
+func (a MempoolResult) AsCommitError() CommitResult {
+	if a.Err == nil {
+		panic("failed assertion: AsCommitError can onyl be called on errors")
```

Comment:
:policeman:
We should not panic in a public API.
Maybe instead a `*CommitResult` should be returned and if `a.Err == nil` then return `nil`.

Reply:
I mean, this is really a programmer assertion.
Not something that would be triggered by network error, disk fault, etc.

I would rather it panic than fail silently.

Reply:
Ok.
I wonder if this is needed at all. I expect that the user of the client will receive `MempoolResult` as an operation result and check `if res.Err != nil { ... }`. Is there a need for this method at all?